### PR TITLE
Integrate with EstablishClaim API

### PR DESCRIPTION
**Pull request by joofsh**
This sets up the API integrations for fetching veteran information (BGS) and creating an EP (VBMS) 

connects #417 

Note: Will have merge conflicts. There is overlap with the work done in https://github.com/department-of-veterans-affairs/caseflow/pull/657 . Will wait for that to be merged first

CC @mdbenjam for awareness since this overlaps with some of your code

Testing:
- [x] Existing tests pass
- [x] Wrote new tests for Dispatch service
- [x] Manual testing of API integration against UAT (via Shane)

### CODE REVIEW

File: app/services/appeal_repository.rb [hunk 1]
```diff
@@ -127,18 +132,37 @@ def self.dateshift_to_utc(value)
...
-    # VBMS.api_call_go!(claim)
+
+  def self.establish_claim!(appeal:, claim:)
+    @vbms_client ||= init_vbms_client
+
+    raw_veteran_record = BGSService.new.fetch_veteran_info(appeal.vbms_id)
+
+    # Reduce keys in raw response down to what we specifically need for
+    # establish claim
+    veteran_record = parse_veteran_establish_claim_info(raw_veteran_record)
```
Comment (shanear) [inline app/services/appeal_repository.rb]: This is the part where bad stuff can happen. Its possible we may want to do some more validation here that things like `first_name`, `last_name`, and `ssn` exist. Thoughts?

File: app/services/dispatch.rb [hunk 2]
```diff
@@ -1,24 +1,71 @@
...
+  def initialize(claim:, task:)
+    # Claim info passed in via browser EP form
+    @base_claim = claim.symbolize_keys
+
+    # Full claim with merged in dynamic and default values
+    @claim = default_claim_values.merge(dynamic_claim_values).merge(@base_claim)
+    @task = task
+  end
+
+  def claim_valid?
```
Comment (shanear) [inline app/services/dispatch.rb]: `ActiveModel::Validators` might come in handy here and make this whole part a little cleaner and more "railsy" ✨ 

http://api.rubyonrails.org/classes/ActiveModel/Validations.html

File: client/app/containers/EstablishClaimPage/EstablishClaim.jsx [hunk 3]
```diff
@@ -34,15 +40,15 @@ export default class EstablishClaim extends BaseForm {
             dateValidator()
           ]
         ),
-        gulfWar: new FormField(false),
-        modifier: new FormField(Form.MODIFIER_OPTIONS[0]),
+        gulfWarRegistry: new FormField(false),
```
Comment (shanear) [inline client/app/containers/EstablishClaimPage/EstablishClaim.jsx]: ❤️ the more descriptive names

File: client/app/util/DateUtil.js [hunk 4]
```diff
@@ -1,10 +1,16 @@
 const ZERO_INDEX_MONTH_OFFSET = 1;
 
+// This will leftpad a day or month so 1/3 becomes 01/03
+export const leftPadDate = (value) => {
```
Comment (shanear) [inline client/app/util/DateUtil.js]: I think mark might have this in his PR too lol

File: app/services/dispatch.rb (hunk 2 above)
Comment (joofsh) [inline app/services/dispatch.rb]: Nice, that sounds much cleaner 👍

File: app/services/appeal_repository.rb (hunk 1 above)
Comment (joofsh) [inline app/services/appeal_repository.rb]: let's discuss in person. I think validations are good but I don't have a good grasp of the scenarios you're concerned about

File: spec/rails_helper.rb [hunk 5]
```diff
@@ -132,6 +133,8 @@ def create_tasks(count, opts = {})
   end
   config.before(:all) { User.unauthenticate! }
 
+  config.after(:each) { Timecop.return }
```
Comment (shanear) [inline spec/rails_helper.rb]: :heart:

Comment (shanear) [inline spec/rails_helper.rb]: :heart:

**Pull request merged**